### ISSUE TITLE
in the current implementation, pinhole images can be scaled according to the resolution parameter, but fisheye images don't have the same scaling mechanism applied.

### ISSUE THREAD

**Issue by zrui22** [denghilbert/Bundle-Adjusting-Gaussian-Splatting#10]
I've tried to modify the loadCam function in utils/camera_utils.py to apply the same resolution scaling logic to fisheye images:

if os.path.exists(ori_path.split('images')[0] + 'fish/images' + ori_path.split('images')[1]):  
    fish_gt_image = Image.open(ori_path.split('images')[0] + 'fish/images' + ori_path.split('images')[1])  
    orig_fish_w, orig_fish_h = fish_gt_image.size  
      
    # Apply the same scaling logic as for original images  
    if args.resolution in [1, 2, 4, 8]:  
        fish_size = (round(orig_fish_w / (resolution_scale * args.resolution)),   
                    round(orig_fish_h / (resolution_scale * args.resolution)))  
    else:  # Apply the same float scaling logic  
        if args.resolution == -1:  
            if orig_fish_w > 1600:  
                global_down = orig_fish_w / 1600  
            else:  
                global_down = 1  
        else:  
            global_down = orig_fish_w / args.resolution  
          
        scale = float(global_down) * float(resolution_scale)  
        fish_size = (int(orig_fish_w / scale), int(orig_fish_h / scale))  
      
    # Resize the fisheye image  
    fish_gt_image = fish_gt_image.resize(fish_size)  
    fish_gt_image_resolution = (3, fish_gt_image.size[1], fish_gt_image.size[0])

After implementing this change, I'm getting poor results with significant distortion in the rendered images. I suspect there might be additional components that need adjustment to properly handle the scaled fisheye images.

Questions
Is my approach to scaling fisheye images correct, or is there a better way to implement this feature?

What other components need to be adjusted when scaling fisheye images? Specifically:

Does the iResNet initialization need to be modified to handle scaled coordinates?
Should the flow_scale parameter be adjusted based on the new scaling logic?
Are there any learning rate or optimization parameters that need adjustment?
Are there recommended parameter settings for control_point_sample_scale, flow_scale, and iresnet_lr when working with scaled fisheye images?

**Comment by denghilbert** [comment]
Hi, thanks for your interest in this project. When you say scaling, I assume you're referring to a larger FOV. Theoretically, it requires a larger flow_scale factor, but the learning rate or other optimization parameters should be fine. control_point_sample_scale controls the density of the explicit grid in the hybrid field. flow_scale defines the distortion boundary you wish to cover.

However, all of the above applies to a single plane. If you truly want to handle a larger FOV (close to 180°), you should use our cubemap module for rasterization. The distortion can still be modeled with a single hybrid neural network, but with cubemap rasterization. Some data loading may be changed manurally. Make sure you load correct initial distortion paramters, focal, and poses.

**Comment by zrui22** [comment]
Hi, thanks again for your help!

Just to clarify: my intention is not to increase the field of view. What I’m doing is downsampling the original 4K fisheye images to 1.6K resolution, mainly to reduce memory usage and ensure consistent input resolution across scenes.

To do this, I modified the loadCam function to apply the same scaling logic used for regular perspective images to the fisheye images. However, after doing so, the rendered fisheye outputs appear distorted or misaligned.

So my question is:

1.When I downsample fisheye images like this, which components need to be adjusted accordingly? For example:

Should I scale the intrinsics (focal length, principal point)?

Does the flow_scale parameter need to be updated?

Do I need to change any iResNet-related logic (e.g. coordinates or initialization)?

2.Are there recommended settings for flow_scale, control_point_sample_scale, or learning rates when using lower-resolution fisheye inputs?

Thanks a lot!

**Comment by denghilbert** [comment]
Sorry for the late response.

When you downsample fisheye images, the focal length should remain unchanged since it's a property of the camera. Naturally, the FOV also remains the same. So, when you downsample the image, you must ensure that the FOV and focal length stay constant, while the height and width change.

As I mentioned, since the intrinsic properties of the camera do not change, the hyperparameters used for reconstruction should still work. However, since you're training on a lower resolution, I can't be 100% sure that the given hyperparameters are the best choice.

Thanks.
